prime-router/src/main/kotlin/azure/ValidateFunction.kt

```diff
@@ -39,35 +38,28 @@ class ValidateFunction(
 ) : Logging, RequestFunction(workflowEngine) {
 
     /**
-     * entry point for the /validate endpoint, which validates a potential submission without writing
+     * entry point for the /validateWithClient endpoint, which validates a potential submission without writing
      * to the database or storing/sending a file
      */
-    @FunctionName("validate")
+    @FunctionName("validateWithClient")
     @StorageAccount("AzureWebJobsStorage")
-    fun run(
+    fun validateWithClient(
         @HttpTrigger(
-            name = "validate",
+            name = "validateWithClient",
```

Comment:
Why not just detect what parameter is being passed and run the proper validation? You should not have different API endpoints just to have different parameters.

Reply:
Thank you for your comment! If we have a single endpoint, then the user-facing documentation will look like so:

![Screenshot 2023-01-25 at 12 58 39 PM](https://user-images.githubusercontent.com/118766341/214658236-fa624b8a-a20c-4ee7-a6b4-73d56ff993e6.png)

Notice, according to the docs, none of the parameters are NOT required, but actually they are. Passing no params is NOT valid, and there are special combos that are accepted:

1. Client
2. SchemaName + format (these params cannot be provided without the other)

The only way this would be clear to the user is if we explicitly added additional documentation specifying the valid combinations. Breaking it out into separate endpoints solved that:

![Screenshot 2023-01-25 at 1 07 52 PM](https://user-images.githubusercontent.com/118766341/214659931-e434c3ba-7f47-4e57-a90a-7351a4288fc4.png)

Another idea may be to use path/route parameters instead, so like:

1. api/validate/{client}
2. api/validate/{schemaName}/{format}

But I think this is usually considered bad practice because route parameters should identify a resource in the system, for example:

1. /users/{id}
2. /organizations/{orgId}/members/{memberId}

In our case, the route parameters would not identify a resource, they would just be placing what are conceptually query parameters into the route parameters.

**Compromise?**

Would something like the following be better?

1. api/validate/withClient (pass client in header, like we do today)
2. api/validate/withSchema?schemaName="NAME&format="FORMAT"

Any additional thoughts or thoughts to what I said? I think this is a discussion worth having and I am willing to update to whatever makes most sense.

Reply:
Perhaps I can do something like the above but manually? Let me see what I can cook up...

Reply:
Okay, I combined the endpoints and manually updated the yml to look like so:

![Screenshot 2023-01-25 at 2 19 39 PM](https://user-images.githubusercontent.com/118766341/214680855-180f2908-5fc6-4627-ab88-035b4a902a35.png)

probably should have done this in the first place, but didn't know I could put whatever in the path in yml

Reply:
Okay so turns out this is invalid openAPI, so I updated per the openAPI spec:

![Screenshot 2023-01-25 at 3 28 25 PM](https://user-images.githubusercontent.com/118766341/214695157-da89eb27-9744-4d30-935d-7592ee1425b3.png)

Doing it this way is what I was trying to avoid in the first place but:
1. It's the recommended workaround here https://swagger.io/docs/specification/describing-parameters/
2. I agree we should not have separate endpoints in this case